perf/src/packet.rs

```diff
@@ -61,7 +65,7 @@ impl Packets {
 pub fn to_packets_chunked<T: Serialize>(xs: &[T], chunks: usize) -> Vec<Packets> {
     let mut out = vec![];
     for x in xs.chunks(chunks) {
-        let mut p = Packets::default();
+        let mut p = Packets::with_capacity(chunks);
```

Comment:
Shouldn't it be `Packets::with_capacity(x.len())`?

Reply:
Oh, yeah, if the `xs` doesn't divide by `chunks`. The last element can still be overallocated. :)

Reply:
Yea, probably not a big deal either way, but might as well size to fit :)